compiler/test/dotty/tools/dotc/printing/SyntaxHighlightingTests.scala

```diff
@@ -0,0 +1,36 @@
+package dotty.tools.dotc.printing
+
+import dotty.tools.io.JFile
+import org.junit.Test
+import org.junit.Assert.fail
+
+import scala.io.Source
+
+/** Runs all tests contained in `compiler/test-resources/printing/`
+  * To check test cases, you can use "cat" or "less -r" from bash*/
```

Comment:
Feature request: add a mode to generate the test files from the syntax highlighter output, this would be useful since you cannot just copy-paste the highlighted output when the test files need to be updated.

Reply:
```scala
  private def testFile(f: JFile): Unit = {
    val linesIt = Source.fromFile(f).getLines()
    val input = linesIt.takeWhile(_ != "result:").mkString("\n")
    val expectedOutput = linesIt.mkString("\n")
    val actualOutput = SyntaxHighlighting(input).mkString

    if (expectedOutput != actualOutput) {
      println("Expected output:")
      println(expectedOutput)
      println("Actual output:")
      println(actualOutput)

      if (f.getName == "nameOfFileToConvertToATest") generateTestFile()

      fail(s"Error in file $f, expected output did not match actual")
    }

    def generateTestFile(): Unit = {
      val path = "compiler/test-resources/printing/" + f.getName
      new PrintWriter(path) {
        write(input + "\nresult:\n" + actualOutput)
        close()
      }
      println(s"Test file for ${f.getName} has been generated")
    }
  }
```

Originally I used a simple function which wrote to a file in the project's root folder and copied the content after "result:". 
This is more convenient. However I'm not sure if it needs explaining and if yes how much. 
(it works on a plain scala file and on a changed test file as well)